fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java

```diff
@@ -571,6 +572,13 @@ public class SessionVariable implements Serializable, Writable, Cloneable {
     @VariableMgr.VarAttr(name = PIPELINE_DOP)
     private int pipelineDop = 0;
 
+    /*
+     * The maximum pipeline dop limit which only takes effect when pipeline_dop=0.
+     * This limitation is to avoid the negative overhead caused by scheduling on super multi-core scenarios.
+     */
+    @VariableMgr.VarAttr(name = MAX_PIPELINE_DOP)
+    private int maxPipelineDop = 64;
```

Comment:
Had better make it invisible since most users don't need to set it ? Or make it a global config ?

Reply:
> Had better make it invisible since most users don't need to set it ? Or make it a global config ?

Although most users do not need to set it, if users are deploying on ultra-multi-core machines, they should know that this variable exists. If it is made into a global configuration, it is not as convenient to modify as a session variable.